lib/datadog/opentelemetry/metrics.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module Datadog
+  module OpenTelemetry
+    module Metrics
+      module Initializer
+        module_function
+
+        def initialize!(settings)
+          return false unless settings.opentelemetry.metrics.enabled
+
+          unless defined?(::OpenTelemetry::SDK)
+            Datadog.logger.warn('OpenTelemetry metrics requested but opentelemetry-sdk gem is not available')
+            return false
+          end
+
+          unless defined?(::OpenTelemetry::SDK::Metrics)
+            Datadog.logger.warn('OpenTelemetry metrics requested but opentelemetry-metrics-sdk gem is not available')
+            return false
+          end
+
+          configure_metrics_sdk(settings)
+          true
+        rescue StandardError => e
+          Datadog.logger.error("Failed to initialize OpenTelemetry metrics: #{e.message}")
+          Datadog.logger.debug(e.backtrace.join("\n")) if Datadog.configuration.diagnostics.debug
+          false
+        end
+
+
+        private
+
+        module_function
+
+        def configure_metrics_sdk(settings)
+          require 'opentelemetry/sdk'
+          require 'opentelemetry-metrics-sdk'
+
+          current_provider = ::OpenTelemetry.meter_provider
+          if current_provider.is_a?(::OpenTelemetry::SDK::Metrics::MeterProvider)
+            current_provider.shutdown
+          end
+
+          resource = create_resource(settings)
+          provider = ::OpenTelemetry::SDK::Metrics::MeterProvider.new(resource: resource)
+          configure_metric_reader(provider, settings)
+          ::OpenTelemetry.meter_provider = provider
+        end
+
+        def create_resource(settings)
+          resource_attributes = {'host.name' => Datadog::Core::Environment::Socket.hostname}
+
+          settings.tags&.each do |key, value|
+            otel_key = case key
+            when 'service' then 'service.name'
+            when 'env' then 'deployment.environment'
+            when 'version' then 'service.version'
+            else key
+            end
+            resource_attributes[otel_key] = value
+          end
+
+          resource_attributes['service.name'] = settings.service if settings.service
+          resource_attributes['deployment.environment'] = settings.env if settings.env
+          resource_attributes['service.version'] = settings.version if settings.version
```

Comment:
`settings.tags` will override `settings.service` ,`.env`, and `.version` when applicable:
https://github.com/DataDog/dd-trace-rb/blob/0bf57812e13db0f25a13bb51f67eedee7164664f/lib/datadog/core/configuration/settings.rb#L718-L721

This means that `when 'service' then 'service.name'` will always be overwritten by `resource_attributes['service.name'] = settings.service if settings.service`.

I see that we still want to process other keys (`else key`), but maybe a better way to write this could be:
```ruby
          resource_attributes.merge(settings.tags.reject do |key, value|
            key == 'service' || key == 'env' || key == 'version'
          end) if settings.tags
```

Reply:
In python we keep all the values in settings.tags and and env/version/service in sync